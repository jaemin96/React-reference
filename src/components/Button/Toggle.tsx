import React from 'react';

interface ToggleProps {
  isToggled?: boolean;
}

export const Toggle = ({ isToggled = false }: ToggleProps) => {
  const [visible, setVisible] = React.useState(isToggled);
  const onOff = visible ? 'toggle--on' : 'toggle--off';

  const handleVisible = () => {
    setVisible(!visible);
  };

  return (
    <div className={['toggle', `${onOff}`].join(' ')} onClick={handleVisible} data-testid="toggle">
      <div className="toggle--circle"></div>
    </div>
  );
};
