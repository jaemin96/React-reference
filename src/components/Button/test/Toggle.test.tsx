import { Toggle } from '../Toggle';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Toggle', () => {
  test('render toggle button', () => {
    render(<Toggle />);
    const toggleElement = screen.getByTestId('toggle');
    expect(toggleElement).toBeInTheDocument();
  });

  test('check init status', () => {
    render(<Toggle />);
    const toggleElement = screen.getByTestId('toggle');
    expect(toggleElement).toHaveClass('toggle--off');
  });

  test('check toggle button status - ON', () => {
    render(<Toggle isToggled={true} />);
    const toggleElement = screen.getByTestId('toggle');
    expect(toggleElement).toHaveClass('toggle--on');
  });

  test('check toggle button status - OFF', () => {
    render(<Toggle isToggled={false} />);
    const toggleElement = screen.getByTestId('toggle');
    expect(toggleElement).toHaveClass('toggle--off');
  });

  test('trans status - Off to On', () => {
    render(<Toggle isToggled={false} />);
    const toggleElement = screen.getByTestId('toggle');
    fireEvent.click(toggleElement);
    expect(toggleElement).toHaveClass('toggle--on');
  });

  test('trans status - On to Off', () => {
    render(<Toggle isToggled={true} />);
    const toggleElement = screen.getByTestId('toggle');
    fireEvent.click(toggleElement);
    expect(toggleElement).toHaveClass('toggle--off');
  });

  test('snapshot', () => {
    const { asFragment } = render(<Toggle data-testid="toggle" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
