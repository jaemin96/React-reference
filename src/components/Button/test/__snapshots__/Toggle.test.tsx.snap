// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle snapshot 1`] = `
<DocumentFragment>
  <div
    class="toggle toggle--off"
    data-testid="toggle"
  >
    <div
      class="toggle--circle"
    />
  </div>
</DocumentFragment>
`;
